app/screens/channel/channel_post_list/channel_post_list.js

```diff
@@ -54,19 +54,27 @@ export default class ChannelPostList extends PureComponent {
         const {postIds: nextPostIds} = nextProps;
 
         let visiblePostIds = this.state.visiblePostIds;
+        const channelSwitch = nextProps.channelId !== this.props.channelId;
 
         if (nextPostIds !== this.props.postIds || nextProps.postVisibility !== this.props.postVisibility) {
             visiblePostIds = this.getVisiblePostIds(nextProps);
         }
 
         this.setState({
             visiblePostIds,
-        });
+            loading: channelSwitch,
+        }, () => InteractionManager.runAfterInteractions(() => {
+            if (channelSwitch) {
+                this.setState({loading: false});
+            }
+        }));
     }
 
     componentDidUpdate(prevProps) {
-        if (prevProps.channelId !== this.props.channelId && tracker.channelSwitch) {
-            this.props.actions.recordLoadTime('Switch Channel', 'channelSwitch');
+        if (prevProps.channelId !== this.props.channelId) {
+            if (tracker.channelSwitch) {
```

Comment:
can we move this again to the way it was? I don't see a need for changing it now